### ISSUE TITLE
[Bug]: SubNavigation should be rendered only if needed. 

### ISSUE THREAD

**Issue by alessioventuriniAND** [ONSdigital/design-system#2976]
### What happened?

The current implementation of the subnavigation within the navigation component has the below code  : 

```nunjucks
<nav class="ons-navigation ons-navigation--sub ons-u-d-no@xxs@l{{ ' ' + params.navigation.subNavigation.classes if params.navigation.subNavigation.classes }}" id="{{ params.navigation.subNavigation.id }}" aria-label="{{ params.navigation.subNavigation.ariaLabel | default("Section menu") }}" data-analytics="header-section-navigation">
            <div class="ons-container ons-container--gutterless@xxs@l{{ ' ons-container--full-width' if params.fullWidth or params.navigation.fullWidth }}{{ ' ons-container--wide' if params.wide or params.navigation.wide }}">
                <ul class="ons-navigation__list ons-navigation__list">
                    {% for item in params.navigation.subNavigation.itemsList %}
                        <li class="ons-navigation__item {{ item.classes }}{{ ' ons-navigation__item--active' if (item.url == params.navigation.currentPath) or (params.navigation.currentPath is not string and item.url in params.navigation.currentPath) }}">
                            <a class="ons-navigation__link ons-navigation__link" href="{{ item.url }}" {% if item.ariaLabel %}aria-label="{{ item.ariaLabel }}" {% endif %} {% if item.id %}id="{{ item.id }}" {% endif %}>
                                {% if item.title %}{{ item.title }}
                                {% elif item.text %}{{ item.text }}
                                {% endif %}
                            </a>
                        </li>
                    {% endfor %}
                </ul>
            </div>
        </nav>


``` 

This renders a horizontal subnavigation which then disappears on mobile view and is replaced by a dropdown button with links to the subnavigation pages. 

We should add a parameter that allow the user to disable the horizontal subnavigation as even though the user can hide this navigation through classes the html is still in the page. 

### What was the expected outcome?

To be able to disable the horizontal navigation completely.

### How is this problem impacting your service?

Unnecessary html in the page.  

### What list of steps can be followed to reproduce the bug or issue?

you can look at the IDS website and here : https://integrateddataservice.gov.uk/about-the-integrated-data-service

Look for header-section-navigation and you will find a nav element that is never displayed. 

### Example

![Screenshot 2023-12-11 at 15 41 50](https://github.com/ONSdigital/design-system/assets/112873190/d99fb7c0-dd48-4055-a06f-4b21a1436e70)

### What version of the ONS Design System are you using?

66.0.2

**Comment by rmccar** [comment]
Blocked: need more discussion with Alessio. Not sure we need to do this because it could cause issues when page is resized

**Comment by alessioventuriniAND** [comment]
This is now unblocked but please reach out to Alessio once you pick this up to clarify any questions before starting.

**Comment by alessioventuriniAND** [comment]
Once this fix is implemented we also need to raise another ticket for Oscar to update the documentation of the header component to include details about the subnavigation parameter within navigation.
